agent_token_test.go

```diff
@@ -0,0 +1,111 @@
+package tfe
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAgentTokensList(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	apTest, apTestCleanup := createAgentPool(t, client, nil)
+	defer apTestCleanup()
+
+	agentToken1, agentToken1Cleanup := createAgentToken(t, client, apTest)
+	defer agentToken1Cleanup()
+	agentToken2, agentToken2Cleanup := createAgentToken(t, client, apTest)
+	defer agentToken2Cleanup()
+
+	t.Run("with no list options", func(t *testing.T) {
+		tokenlist, err := client.AgentTokens.List(ctx, apTest.ID)
+		require.NoError(t, err)
+		assert.Contains(t, tokenlist.Items, agentToken1)
+		assert.Contains(t, tokenlist.Items, agentToken2)
```

Comment:
These fail; based on the test output it looks like some types are mismatched so they aren't treated equal. Looks like the synonymous test for user tokens might be helpful here.

Reply:
First, it's about the token itself: it's returned at generating only and not if you read the token afterward.
And second, I think the list is returning correct IDs, and there seem to be differences in the timestamp representation.

Anyway, what I didn't get (while copy/pasting and search/replace most of the code 😉 ) is if I really need the AgentTokenList function? While implementing the tfe_agent_token resource, I didn't need it anywhere.
Should I remove it or is there any other need for it?

Reply:
Ah right, forgot about the token value being omitted on the read afterward.

You don't need the list function for your purposes in the provider resource but for the client in general but it's a perfectly valid and supported endpoint otherwise for the client, might as well keep it.